addons/crm/tests/test_crm_lead_multicompany.py

```diff
@@ -62,6 +62,41 @@ def test_lead_mc_company_computation(self):
         self.assertEqual(lead_team_no_company.company_id, self.company_2)
         self.assertEqual(lead_team_no_company.team_id, self.team_company2)
 
+    @users('user_sales_manager_mc')
+    def test_lead_mc_company_computation_with_restricted_allowed_company_ids(self):
+        """ Check that the computed company is allowed (so contained in self.env.companies)
+        The user is logged in the company_main even though his default company is company_2
+        """
+
+        self_with_2nd_co = self.env['crm.lead'].with_context(allowed_company_ids=[self.company_main.id])
```

Comment:
``LeadCompanyMain``? Not sure to understand this "2nd_co" ...

Reply:
`self_with_2nd_co` is to simulate the user logged in with is non default company. Not sure what's a better name for it, any ideas ?